crypto/bn/bn_gcd.c

```diff
@@ -534,6 +534,34 @@ BIGNUM *BN_mod_inverse(BIGNUM *in,
     return rv;
 }
 
+/*
+ * The numbers a and b are coprime if the only positive integer that is a
+ * divisor of both of them is 1.
+ * i.e. gcd(a,b) = 1.
+ *
+ * Coprimes have the property: b has a multiplicative inverse modulo a
+ * i.e there is some value x such that bx = 1 (mod a).
+ *
+ * Testing the modulo inverse is currently much faster than the constant
+ * time version of BN_gcd().
+ */
+int BN_are_coprime(BIGNUM *a, const BIGNUM *b, BN_CTX *ctx)
+{
+    int ret = 0;
+    BIGNUM *tmp = BN_new();
```

Comment:
Use BN_CTX_start(ctx), BN_CTX_get(ctx)?

Reply:
I thought about doing that, is it worth it for the setup involved?

Reply:
Using the CTX could easily be a freebie.  No allocations most of the time.
So, yes probably worthwhile.

Reply:
Ok Done, I have chosen not to check and do the ctx alloc... I know we do the alloc in other places, but this seems like a place where it would be bad to be doping this if used in loops.

Reply:
In a loop, it will be free after the first iteration.....

Reply:
I'm not sure I really understand your comments. As I understand it, the point of BN_CTX_start() is to avoid overhead of BN_new(). I understand that this function is called in a loop, so we want to avoid the BN_new() call. BN_CTX_start() creates a new stack frame, BN_CTX_end() cleans up that stack frame. The documentation for those functions can probably be improved.

Reply:
What I meant was that I dont do the ctx alloc if it is NULL. i.e.

if (ctx == NULL)
   ctx = BN_CTX_new();

Reply:
Pauli I dont understand your comment.. Do you require me to do something else here?

Reply:
I'm not sure that not supporting NULL for the ctx is a good idea. I assume all our other APIs support NULL for it.

Reply:
No, nothing from my perspective.